scripts/cplane/cluster_init.sh

```diff
@@ -0,0 +1,351 @@
+#!/usr/bin/env bash
+
+#
+# Script to Initialize the Control Plane
+#
+# There is a lot in this script. If you're learning, don't be put off
+# look below for "SCRIPT ESSENTIALS" and it shows the small set of
+# important functions, which are:
+#
+# 1. Grab the kubernetes service images
+# 2. Init the service, specifying the base CIDR
+# 3. Copy the kube config to the user directory
+# 4. Install the CNI service
+#
+# Unicode Character icons (https://www.compart.com/en/unicode) for pretty scripts
+#
+# 😄 Generic Info
+# ☸️ Kubernetes
+# ✨ Perform Magic
+# ⚙️ Setting something
+# 🚜 Image pull
+# 🛠 Tools / Install
+# 🔍 Get Info or Data or Config
+# ✅ Good Result
+# ❌ Bad Result
+# ⬆ Up Arrow
+# ⬇ Down Arrow
+# ❓ Question / Unknown
+# 🤷 Something missing
+# 🔥 Creating something
+# 👍 Startup
+# ⏳ Waiting
+# 🔄 Restarting
+#
+
+# These variables define the Heart of the Cluster: CP ID and IP base for Pods
+#
+# Strip off errant 'localhost-y' reference that get created via vagrant
+API_SERVER_IP=$(echo $(hostname -i | sed -E 's/127\.0\.[0-9]+\.[0-9]+//g'))
+POD_BASE_CIDR=10.244.0.0 # Base address for pods
+
+# Hey, howzitgoing
+function welcome_msg {
+    echo "Kubernetes Control Plane / Cluster Init"
+    echo ""
+    echo "POD_BASE_CIDR: '${POD_BASE_CIDR}'"
+    echo "API_SERVER_IP: '${API_SERVER_IP}'"
+    echo ""
+}
+
+
+# Quick check to see if Kubernetes utilities are installed
+function kube_sanity_check {
+    # Are Kubernetes tools installed?
+    echo "🔍 Check for kubernetes tools on the machine"
+
+    kube_tools="kubelet kubeadm kubectl"
+    for ktool in $(echo ${kube_tools}); do
+        if ! command -v ${ktool} >/dev/null 2>&1; then
+            echo "🤷 utility '${ktool}' not found"
+            ktoolmissing=y
+        fi
+    done
+
+    if [ "${ktoolmissing}" == "y" ] ; then
+        echo "❌ one ore more kubernetes utilities missing; install before continuing"
+        exit 1
+    fi
+}
+
+
+# Verify that the api server is up and running ?
+# or all are down
+#
+# Param 1 - "up" or "down"
+# Param 2 - optional - "retry" for a retry loop, useful
+#           for when services are taking time to come up/down
```

Comment:
Long function, which is not essential but a nice "sanity check", determining the state of the kubernetes services before doing anything.  We want them to be "up" to do certain work, or "down" in case we're doing the cluster init.